manager/resource.go

```diff
@@ -278,18 +288,26 @@ func (rm *ResourceManager) decrementRefCounters(p *v1.Pod) {
 	for _, c := range p.Spec.Containers {
 		for _, e := range c.Env {
 			if e.ValueFrom != nil && e.ValueFrom.ConfigMapKeyRef != nil {
-				rm.configMapRef[e.ValueFrom.ConfigMapKeyRef.Name]--
+				configMapKey := rm.getStoreKey(p.Namespace, e.ValueFrom.ConfigMapKeyRef.Name)
+				rm.configMapRef[configMapKey]--
 			}
 
 			if e.ValueFrom != nil && e.ValueFrom.SecretKeyRef != nil {
-				rm.secretRef[e.ValueFrom.SecretKeyRef.Name]--
+				secretKey := rm.getStoreKey(p.Namespace, e.ValueFrom.SecretKeyRef.Name)
+				rm.secretRef[secretKey]--
 			}
 		}
 	}
 
 	for _, v := range p.Spec.Volumes {
 		if v.VolumeSource.Secret != nil {
-			rm.secretRef[v.VolumeSource.Secret.SecretName]--
+			secretKey := rm.getStoreKey(p.Namespace, v.VolumeSource.Secret.SecretName)
+			rm.secretRef[secretKey]--
 		}
 	}
 }
+
+// getStoreKey return the key with namespace for store objects from different namespaces
+func (rm *ResourceManager) getStoreKey(namespace, name string) string {
+	return namespace + "-" + name
```

Comment:
'-' is a valid character in the name or namespace. suggest to change it to '_' to avoid conflict